src/main/java/org/sonar/plugins/buildbreaker/QualityGateBreaker.java

```diff
@@ -186,7 +191,8 @@ String getAnalysisId(WsClient wsClient, String ceTaskId) {
             throw new IllegalStateException(
                 "Report processing did not complete successfully: " + taskStatus);
         }
-      } catch (IOException | InterruptedException e) {
+      } catch (IOException | InterruptedException e) { // NOSONAR: We have to treat a premature
```

Comment:
I think `// NOSONAR` violates [RSPEC-139](https://rules.sonarsource.com/java/tag/convention/RSPEC-139) and therefore I suggest using `@SupressWarnings` instead.

Reply:
Good point. While I'm sure Sonar wouldn't complain about its own marker comments, it's actually cleaner to use `SuppressWarnings` on the method: both for the awkward line-break and because it's sensible to treat a thread interrupt on a method level.